packages/react/src/components/form/flow.ts

```diff
@@ -0,0 +1,167 @@
+import { Cancel, LogIn, LoginOptions, Retry } from "../../domain/types";
+
+export interface InitialState {
+  status: "initial";
+  logIn: (options: LoginOptions) => void;
+}
+
+export interface AuthenticatingState {
```

Comment:
@vincenzoiozzo Each state is referenced in the `FlowState` union type. `status` is used as the discriminating property so we can easily check which type we're looking at based on that value.

States have context and actions - context is the related data we guarantee to exist at that time, while actions are the functions that can be called while in that state.

On the consumer side this makes it easy to call the proper methods, because after testing the state type using `status` you can safely call these actions - TypeScript will tell you what is available.